templates/repo/settings/options.tmpl

```diff
@@ -67,12 +67,30 @@
 
 		</div>
 
-		{{if .MirrorsEnabled}}
+		{{if or (not (and .DisableNewPullMirrors .DisableNewPushMirrors)) .Repository.IsMirror}}
```

Comment:
I'll write this in free text:

If the repository is a mirror OR new pull mirrors and new push mirrors aren't allowed, then the repository's settings will be shown to the user.

Otherwise, changing the mirror settings for repositories that were created *before* hypothetically disabling the mirrors would not be allowed. `.MirrorsEnabled` effectively disables new pull mirrors and new push mirrors, so I thought that this condition would be reasonable.

If I am not wrong (there's a chance that I am, I don't remember if I managed to verify this or not), disabling mirrors removes the mirror status. This needs to be checked again.

Reply:
> If I am not wrong (there's a chance that I am, I don't remember if I managed to verify this or not), disabling mirrors removes the mirror status. This needs to be checked again.

What I said earlier seems to be wrong (Sorry, I "bruteforced" this change until it worked and forgot many of the details right afterwards).

Should the settings remain modifiable from the web UI if the repository is marked as a mirror and mirrors are disabled?

Reply:
If it's a mirror repository, it will be displayed. Even if the mirror is disabled after the mirror repository is created. Although the mirroring action will be stopped because the mirroring is disabled. So maybe it could still be editable but maybe we should have a remind that mirroring is disabled, this mirror repository will not be updated.

Reply:
Excellent idea, will implement later.

Reply:
Disabling .MirrorsEnabled results in the change - implemented as you described it - not being visible at all. I can't spot any relevant code excerpts and believe that my condition is otherwise correct.

![image](https://github.com/go-gitea/gitea/assets/30193966/841b30f9-dddb-4fe0-a032-073ab2174d36)